src/igvfd/types/file_set.py

```diff
@@ -118,6 +118,94 @@ class CuratedSet(FileSet):
     schema = load_schema('igvfd:schemas/curated_set.json')
     embedded_with_frame = FileSet.embedded_with_frame
 
+    @calculated_property(
+        schema={
+            'title': 'Assembly',
+            'description': 'The genome assembly to which the referencing files in the file set are utilizing (e.g., GRCh38).',
+            'type': 'array',
+            'uniqueItems': True,
+            'items': {
+                'type': 'string'
+            },
+            'notSubmittable': True,
+        }
+    )
+    def assembly(self, request, files):
+        assembly_values = set()
+        if files is not None:
+            for current_file_path in files:
+                file_object = request.embed(current_file_path, '@@object?skip_calculated=true')
+                if file_object.get('assembly'):
+                    assembly_values.add(file_object.get('assembly'))
+        assembly_value_list = list(assembly_values)
+        assembly_value_list.sort()
+        if len(assembly_value_list) == 0:
+            assembly_value_list = None
+        return assembly_value_list
+
+    @calculated_property(
+        schema={
+            'title': 'Transcriptome Annotation',
+            'description': 'The annotation and version of the reference resource.',
+            'type': 'array',
+            'uniqueItems': True,
+            'items': {
+                'type': 'string'
+            },
+            'notSubmittable': True,
+        }
+    )
+    def transcriptome_annotation(self, request, files):
+        annotation_values = set()
+        if files is not None:
+            for current_file_path in files:
+                file_object = request.embed(current_file_path, '@@object?skip_calculated=true')
+                if file_object.get('transcriptome_annotation'):
+                    annotation_values.add(file_object.get('transcriptome_annotation'))
+        annotation_value_list = list(annotation_values)
+        annotation_value_list.sort()
+        if len(annotation_value_list) == 0:
+            annotation_value_list = None
+        return annotation_value_list
+
+    @calculated_property(
+        schema={
+            'title': 'Summary',
+            'type': 'string',
+            'notSubmittable': True,
+        }
+    )
+    def summary(self, request, curated_set_type, files, taxa=None):
+
+        assembly_values_set = set()
+        annotation_values_set = set()
+        if files is not None:
+            for current_file_path in files:
+                file_object = request.embed(current_file_path, '@@object?skip_calculated=true')
+                if file_object.get('assembly'):
+                    assembly_values_set.add(file_object.get('assembly'))
+                if file_object.get('transcriptome_annotation'):
+                    annotation_values_set.add(file_object.get('transcriptome_annotation'))
```

Comment:
I thikn in the summary you can use the "assembly" and "transcriptome_annotation" calculated properties you defined above, not traverse files again

Reply:
I tried that.  The value `None` is always passed.  It seems we can't pass calculated properties as parameters to other calculated properties.  Each calculated property seems to be self contained, and can't have references to other calculated properties.

Reply:
are you sure it is always None, even if the files have assembly?

Reply:
Yes, because the data is being set in the unit test prior to this and it still returns NONE for the other calculated properties.